src/core/api.c

```diff
@@ -631,8 +631,12 @@ MsQuicStreamOpen(
 
     QUIC_CONN_VERIFY(Connection, !Connection->State.Freed);
 
-    if (QuicConnIsClosed(Connection)) {
-        Status = QUIC_STATUS_INVALID_STATE;
+    BOOLEAN ClosedLocally = Connection->State.ClosedLocally;
```

Comment:
Why is `ClosedLocally` cached in a local variable, but `ClosedRemotely` isn't?

Reply:
Because ClosedLocally is read twice. Additionally, this function specifically is not explicitly called on the worker, so theres a definite possibility for 2 reads here to return 2 different values.